### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by LOctavio**
- Remove current_user method in ApplicationController, devise will provide us with one.
- Install and set up devise.
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed passwords should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,16 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  protect_from_forgery with: :exception
+
+  before_action :update_allowed_parameters, if: :devise_controller?
+
```
Comment (Lameck1) [inline app/controllers/application_controller.rb:6]: - [ ] It looks like we are not done with setting up `devise`. Please ensure that all users get authenticated before taking any action in the App by adding this 👇🏿:
  ```ruby
  before_action :authenticate_user!
  ```
  
  With this `devise` will now serve its purpose of securing the app. Currently devise is installed but not properly setup to secure the app.

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      </div>
+    <% end %>
+
+    <div class="actions">
+      <%= f.submit "Log in" %>
+    </div>
+  <% end %>
+
+  <%= render "devise/shared/links" %>
+</div>
```
Comment (Lameck1) [inline app/views/devise/sessions/new.html.erb]: - [ ] As in the instructions, this view is supposed to be modified to match the wireframe below:
  
  image
  
  Currently it is like this 👇🏿:  
  
  image
  
  As you make the change, remember to configure `devise` to only accept username and password for log_in as in the wireframe. Please consider working on this:

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @LOctavio, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb (hunk 2 above)
Comment (Dipeshtwis) [inline app/views/devise/sessions/new.html.erb]: - [ ] I would recommend following the design of login. Currently, your design is not matching with the wireframe.

### Your design 
Screenshot from 2022-01-25 22-26-14


### Given design
Screenshot from 2022-01-25 22-29-02

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (Dipeshtwis) [inline app/controllers/application_controller.rb:6]: - [ ] It looks like we are not done with setting up `devise`. Please ensure that all users get authenticated before taking any action in the App by adding this 👇🏿:

```ruby
before_action :authenticate_user!
```

With this `devise` will now serve its purpose of securing the app. Currently devise is installed but not properly setup to secure the app.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @DyaryRaoof, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
### Hi @LOctavio   🙋‍♀️, 

# STATUS APPROVED ✔

Thanks for making the requested changes 👏 
   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉

## Hi @LOctavio, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**